apis/dns/v1alpha1/dnspolicy_types.go

```diff
@@ -0,0 +1,129 @@
+	// +optional
+	AlternativeNameServerConfig *PolicyAlternativeNameServerConfig `json:"alternativeNameServerConfig,omitempty"`
+
+	// Description: A mutable string of at most 1024 characters associated with this resource for the user's convenience.
+	// Has no effect on the policy's function.
+	// +optional
+	Description string `json:"description"`
+
+	// EnableInboundForwarding: Allows networks bound to this policy to receive DNS queries sent by VMs or applications over VPN connections.
+	// When enabled, a virtual IP address is allocated from each of the subnetworks that are bound to this policy.
+	// +optional
+	EnableInboundForwarding *bool `json:"enableInboundForwarding,omitempty"`
+
+	// EnableLogging: Controls whether logging is enabled for the networks bound to this policy.
+	// Defaults to no logging if not set.
+	// +optional
+	EnableLogging *bool `json:"enableLogging,omitempty"`
+
+	// Networks: List of network names specifying networks to which this policy is applied.
+	// +optional
+	Networks *[]PolicyNetwork `json:"networks,omitempty"`
+}
+
+// The PolicyAlternativeNameServerConfig Sets an alternative name server for the associated networks.
+// When specified, all DNS queries are forwarded to a name server that you choose.
+type PolicyAlternativeNameServerConfig struct {
+
+	// TargetNameServers: Sets an alternative name server for the associated
+	// networks. When specified, all DNS queries are forwarded to a name
+	// server that you choose. Names such as .internal are not available
+	// when an alternative name server is specified.
+	TargetNameServers []PolicyAlternativeNameServerConfigTargetNameServer `json:"targetNameServers"`
+}
+
+// A PolicyAlternativeNameServerConfigTargetNameServer has the below fields.
+type PolicyAlternativeNameServerConfigTargetNameServer struct {
+
+	// ForwardingPath: Forwarding path for this TargetNameServer. If unset or set to DEFAULT, Cloud DNS makes forwarding decisions based on  address ranges; that is, RFC1918 addresses go to the VPC network, non-RFC1918 addresses go to the internet. When set to PRIVATE, Cloud
+	// DNS always sends queries through the VPC network for this target. Possible values:
+	// "default" - Cloud DNS makes forwarding decision based on IP address ranges; that is, RFC1918 addresses forward to the target through the VPC and non-RFC1918 addresses forward to the target through the internet
+	// "private" - Cloud DNS always forwards to this target through the VPC.
+	ForwardingPath *string `json:"forwardingPath,omitempty"`
+
+	// Ipv4Address: IPv4 address to forward to.
+	Ipv4Address string `json:"ipv4Address"`
+}
+
+// A PolicyNetwork struct has the field NetworkURL
+type PolicyNetwork struct {
+
+	// NetworkUrl: The fully qualified URL of the VPC network to bind to.
+	// This should be formatted like https://www.googleapis.com/compute/v1/projects/{project}/global/networks/{network}
+	NetworkURL string `json:"networkUrl"`
```

Comment:
```suggestion
	// This should be formatted like https://www.googleapis.com/compute/v1/projects/{project}/global/networks/{network}
	// +crossplane:generate:reference:type=github.com/crossplane-contrib/provider-gcp/apis/compute/v1beta1.Network
	// +crossplane:generate:reference:extractor=github.com/crossplane-contrib/provider-gcp/apis/compute/v1beta1.NetworkURL()
	NetworkURL string `json:"networkUrl"`
```

We have a concept called cross-resource references that will let you reference a `Network` custom resource instead of having to provide the full URL when you create the custom resource. Just adding the two lines above should be sufficient 🙂